src/screens/reader/utils/sanitizeChapterName.ts

```diff
@@ -0,0 +1,53 @@
+export const sanitizeChapterName = (
+  text: string,
+  showChapterName: boolean | undefined,
+  sourceChapterName: string,
+): string => {
+  let firstLines = text.split(/(.+\n)/gi, 6);
+  const [chapterNumber, chapterName] = takeName(sourceChapterName);
+  const nameOnLine = findChapterName(firstLines, chapterName);
+  if (showChapterName === undefined) {
+    return text;
+  }
+  if (nameOnLine !== -1) {
+    text = deleteTitle(text, firstLines, nameOnLine);
+  }
+  if (showChapterName) {
+    text =
+      '<h3 style="margin-bottom:-20px">' +
+      chapterNumber +
+      '</h3>\n<h2>' +
+      chapterName +
+      '</h2>' +
+      text;
+  }
+  return text;
+};
+
+const deleteTitle = (
+  text: string,
+  lines: Array<string>,
+  nameOnLine: number,
+): string => {
+  let toDeleteText = '';
+  for (let i = 0; i <= nameOnLine; i++) {
+    toDeleteText += lines[i];
+  }
+  return text.replace(toDeleteText, '');
+};
+const findChapterName = (lines: Array<string>, chapterName: string) => {
+  for (let index = 0; index < lines.length; index++) {
+    if (lines[index].includes(chapterName)) {
```

Comment:
Wouldn't passing the entire text and checking if it includes chapterName be same instead of iterating of the lines?

Also, would only checking if the text includes the name be right?

Reply:
It is possible to pass the entire text, however often the Autor puts the title somewhere inside the text, that is why I only take the first few lines. As to why I separate the name, in one of my test chapters the title had a "-" in between the number and the actual name and in the chapter was the title without the "-" resulting in two titles.